app/src/ui/abort-merge/abort-merge-warning.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react'
+import { Button } from '../lib/button'
+import { ButtonGroup } from '../lib/button-group'
+import { Dialog, DialogContent, DialogFooter } from '../dialog'
+import { Dispatcher } from '../../lib/dispatcher'
+import { PopupType } from '../../lib/app-state'
+import { Repository } from '../../models/repository'
+import { abortMerge } from '../../lib/git'
+import { Octicon, OcticonSymbol } from '../octicons'
+
+interface IAbortMergeWarningProps {
+  readonly dispatcher: Dispatcher
+  readonly repository: Repository
+  readonly onDismissed: () => void
+  readonly currentBranchName: string
+  readonly comparisonBranchName: string
+}
+
+const titleString = 'Confirm abort merge'
+const cancelButtonString = 'Cancel'
+const abortButtonString = 'Abort merge'
+
+/**
+ * Modal to tell the user their merge encountered conflicts
+ */
+export class AbortMergeWarning extends React.Component<
+  IAbortMergeWarningProps,
+  {}
+> {
+  /**
+   *  aborte the merge and dismisses the modal
+   */
+  private onSubmit = async () => {
+    await abortMerge(this.props.repository)
```

Comment:
I didn't implement any error handling here because I was thinking this would handle it:

https://github.com/desktop/desktop/blob/b8d305fd2f5b9333032e10abac4e89798ad71766/app/src/lib/dispatcher/error-handlers.ts#L62

Is that assumption wrong? `abortMerge` will throw an error in the event of a failed abort.

We don't have really have a full error flow sketched out, so the only thing I can think of is to make sure the popup state is changed to ensure you're not returned to this popup over and over again? Or is there something else you had in mind?

cc @ampinsk for more thoughts

Reply:
This is missing the plumbing that comes with using `GitStore.performFailableOperation` to wrap an operation and emit any potential error from `GitStore` -> `AppStore` -> `App` -> `Dispatcher.postError`.

Without that, you just get these uncaught errors in the dev tools:

![](https://user-images.githubusercontent.com/359239/47306542-188deb00-d603-11e8-83e7-16ab31d089bb.png)